src/redux/sagas.js

```diff
@@ -47,10 +54,20 @@ function* postSpendingWorker(data) {
   }
 }
 
+function* updateSpendingWorker(action) {
+  try {
+    yield call(spendings.updateSpending, action.payload.id, action.payload.spending);
+    yield put(updateSpendingSuccess(action.payload.id, action.payload.spending));
```

Comment:
Backend should return to the frontEnd updated spending and that spending you should store.

Reply:
> Backend should return to the frontEnd updated spending and that spending you should store.

Is it right?

```
const updateSpending = async (req, res) => {
  try {
    const { ...values } = req.body
    const id = req.params.id
    const spending = await Spending.findByIdAndUpdate(id, `values)`
    res.status(HTTP_STATUS_CODES.OK).send(spending)
  } catch (error) {
    res
      .status(HTTP_STATUS_CODES.INTERNAL_SERVER_ERROR)
      .send({ error: 'Internal Error - Internal server error' })
  }
}
```

Reply:
```
function* updateSpendingWorker(action) {
  try {
    const response = yield call( spendings.updateSpending, action.payload.id,  action.payload.spending );
    yield put(updateSpendingSuccess(action.payload.id, response.data));
  } catch (error) {
    yield put(updateSpendingError());
  }
}
```